app/main.py

```diff
@@ -39,51 +48,257 @@ def execute(self):
+                return result
+            return {"result": evaluate_topology['result'],
+                    "status_code": evaluate_topology['status_code']}
+        except Exception as err:  # pylint: disable=W0703
+            log.info(err)
+        return {"result": "No SDX Topology loaded", "status_code": 401}
+
     @listen_to("kytos/topology.*")
-    def listen_topology(self, event=None):
+    def listen_event(self, event=None):
         """Function meant for listen topology"""
-        f_name = " listen_topology "
-        log.info(f"{HSH}{f_name}get kytos topology {HSH}")
-        event_info = {}  # pylint: disable=W0201
-        if event:
-            allowed_events = [
-                "kytos/topology.switch.enabled",
-                "kytos/topology.switch.disabled",
-                "kytos/topology.link_up",
-                "kytos/topology.link_down",
-            ]
-            if event.name in allowed_events:
-                try:
-                    topology = requests.get(settings.KYTOS_TOPOLOGY_URL).json()
-                    log.info(
-                        f"{HSH}{f_name}constructor post request {HSH}"
-                    )
-                    event_info = {
-                        "event": event,
-                        "kytostopology": topology,
-                    }  # pylint: disable=W0201
-                    try:
-                        requests.post(settings.SDX_CONSTRUCTOR).json(event_info)
-                    except Exception as err:  # pylint: disable=W0703
-                        log.info(f"{HSH}{f_name}constructor post error {HSH}")
-                        log.info(err, event_info)
-                except Exception as err:  # pylint: disable=W0703
-                    log.info(
-                            f"{HSH}{f_name}get kytos topology error {HSH}"
-                    )
-                    log.info(err)
-        return event_info
-
-    @rest("v2/listen_topology", methods=["GET"])
-    def get_listen_topology(self):
-        """consume call listen Topology"""
-        f_name = " get_listen_topology "
+        f_name = " listen_event "
+        log.info(f"{HSH}{HSH}{HSH}{HSH}{HSH}")
+        log.info(f"{HSH}{f_name} {HSH}")
+        log.info(f"{HSH} dir(event): {dir(event)} {HSH}")
+        log.info(f"{HSH} event: {event} {HSH}")
+        log.info(f"{HSH} event.name: {event.name} {HSH}")
+        log.info(f"{HSH} event.content: {event.content} {HSH}")
+        log.info(f"{HSH}{HSH}{HSH}{HSH}{HSH}")
+        if event is not None and self.get_kytos_topology():
+            if event.name in settings.ADMIN_EVENTS:
+                event_type = 1
+            elif event.name in settings.OPERATIONAL_EVENTS and \
+                    event.timestamp is not None:
+                event_type = 2
+            else:
+                return {"event": "not action event"}
+            # open the event shelve
+            with shelve.open("events_shelve") as log_events:
```

Comment:
The goal is to create an event changes tracker that can count from 1 to infinity and the timestamp of the event. I am currently unsure if using mongodb solely for the purpose of tracking the counter is the most efficient approach, or if it may be excessive.

Reply:
I will evaluate putting it on the rabbit mq memory we already implemented or write it down to MongoDB

Reply:
Right, @lmarinve. MongoDB should be able to handle what you need, in prod we have a 3-node cluster, theoretically on average a dozens of thousands of DB ops per sec should be OK, but as a general rule of thumb, as you know, the more we can minimize IO and only really write what we need the better, especially since the DB is also shared over all NApps, although it can also scale horizontally. 

With that said, speaking of minimizing IO, your requirements were OK with only writing to the DB events that you categorized with `ADMIN_EVENTS`, those tend not to be high frequency under normal circumstances, and if even a user were to abuse, in the future NApps decorated endpoints can support rate limit, so later on down that road this shouldn't impact too much your NApp in terms of refactoring and evolving it.